tests/h/accounts/services_test.py

```diff
@@ -0,0 +1,122 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+import mock
+import pytest
+
+from h.accounts.services import UserSignupService, user_signup_service_factory
+from h.models import Activation, Subscriptions, User
+
+
+class TestUserSignupService(object):
+    def test_signup_returns_user(self, svc):
+        user = svc.signup(username='foo', email='foo@bar.com', password='baz')
+
+        assert isinstance(user, User)
+
+    def test_signup_creates_user_in_db(self, db_session, svc):
+        svc.signup(username='foo', email='foo@bar.com', password='baz')
+
+        db_session.commit()
+        db_session.close()
+
+        user = db_session.query(User).filter_by(username='foo').one_or_none()
+
+        assert user is not None
+
+    def test_signup_creates_activation_for_user(self, svc):
+        user = svc.signup(username='foo', email='foo@bar.com', password='baz')
+
+        assert isinstance(user.activation, Activation)
+
+    def test_passes_user_info_to_signup_email(self, svc, signup_email):
+        user = svc.signup(username='foo', email='foo@bar.com', password='baz')
+
+        signup_email.assert_called_once_with(id=user.id,
+                                             email='foo@bar.com',
+                                             activation_code=user.activation.code)
+
+    def test_signup_sends_email(self, mailer, svc):
+        svc.signup(username='foo', email='foo@bar.com', password='baz')
+
+        mailer.send.delay.assert_called_once_with(['test@example.com'],
```

Comment:
It confused me initially that the email was different than that passed to `signup` but I see that this is the hardcoded return value of the `signup_email` fake